test/parallel/test-worker-init-failure.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const child_process = require('child_process');
+
+// Test that workers fail with meaningful error message
+// when their initialization fails.
+
+if (process.argv[2] === 'child') {
+  const { Worker } = require('worker_threads');
+  for (let i = 0; i < 256; ++i) {
+    const worker = new Worker(
+      'require(\'worker_threads\').parentPort.postMessage(2 + 2)',
+      { eval: true });
+    worker.on('message', (result) => {
+      assert.strictEqual(result, 4);
+    });
+    worker.on('error', (e) => {
+      assert(e.message.match(/Worker initialization failure: EMFILE/));
+      assert.strictEqual(e.code, 'ERR_WORKER_INIT_FAILED');
+    });
+  }
+
+} else {
+
+  if (common.isWindows) {
+    common.skip('ulimit does not work on Windows.');
```

Comment:
Maybe instead of skipping the test, just skip the `ulimit` prefix? As written, I don't think the test won't fail if the `error` event never happens, so the test might still be runnable under Windows?

Reply:
> don't think the test won't fail if the error event never happens

I think this is an error in the test and therefore after it is resolved it will no longer be runnable on windows.
Also, it doesn't look like there is any equivalent for ulimit on windows so skip looks like the only way.

Reply:
@Trott , with respect to Windows - is there any merit in running a test that doesn't make any assertion ?